tao_compiler/mlir/xla/ral/context/quantized_gpu_kernel_impl.cc

```diff
@@ -183,6 +183,143 @@ struct GemmScaleState : public Context::Resource {
   std::unordered_map<GemmScaleKey, float, GemmScaleKeyHasher> cache;
 };
 
+struct GemmScaleChannelState : public Context::Resource {
+  std::mutex mu;
+  std::unordered_map<GemmScaleKey, std::vector<float*>, GemmScaleKeyHasher>
+      cache;
+};
+
+// Layout requirement:
+// input: m x k, row major
+// weight: n x k, col major
```

Comment:
the shape of weight is already `nxk`, so should be row major?

Reply:
for weight, `kxn` is row major and `nxk` is colmajor?

Reply:
I see. Maybe just different convention:
`memref<kxn>, row major, strides = [n, 1]`
`memref<kxn>, column major, strides = [1, k]`
`memref<nxk>, column major, strides = [1, n]`
`memref<nxk>, row major, strides = [k, 1]`